Automattic/wp-calypso
# WooCommerce: Add product image management

This PR introduces a product image uploader and allows you to upload multiple product images, as well as remove them. Images are uploaded to the media library.

This is using Calypso's `DropZone` and `FilePicker` components for v1. We should look at using the media modal post v1 (it looked tied to the post-editor/TinyMCE so I went this route for our MVP). 

To test:
* Visit `http://calypso.localhost:3000/store/products/:site/add`.
* Click "Add Images" or drag and drop some images to the upload area.
* Watch as your images upload (spinner/busy state). You should see an instant preview of them.
* Your images should upload (or if an upload fails, an error notice should be displayed).
* You should be able to upload additional images if wanted.
* Click the "x" on an image to remove it from the product.

<img width="773" alt="screen shot 2017-05-30 at 10 54 54 am" src="https://cloud.githubusercontent.com/assets/689165/26597629/79107344-4527-11e7-802f-4007caa8153a.png">

<img width="795" alt="screen shot 2017-05-30 at 10 59 12 am" src="https://cloud.githubusercontent.com/assets/689165/26597642/80a2b306-4527-11e7-916b-2ad28b208868.png">

Closes #13658, #13656.

Related: This works better on the front end with https://github.com/woocommerce/wc-api-dev/pull/1, which removes a dummy placeholder image from the response, and makes setting images without manually setting positions. However, we will need to bump to using v3 with the feature plugin installed. These changes work without that branch, though.
